perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -1579,6 +1660,7 @@ public Member expireMember(PerunSession sess, Member member) throws WrongReferen
 		Status oldStatus = member.getStatus();
 		getMembersManagerImpl().setStatus(sess, member, Status.EXPIRED);
 		member.setStatus(Status.EXPIRED);
+		removeMemberFromParentVos(sess, member);
```

Comment:
Requirements for expired members changed a bit - if a member is expired in a memberVo, he should still be in its parentVos but managed by their lifecycle rules (so we should only remove the memberVo from memberOrganizations attribute and call `extendMembership` method to start managing him by a parentVos expiration lifecycle).

Reply:
Sorry, I just realized this is how you actually implemented :D

Reply:
When I think about it again, if his status changed from INVALID/DISABLED to EXPIRED, he might not exist in parentVos, so we should probably create him there (use a little changed `addMemberToParentVos` instead of `removeMemberFromParentVos`) - what do you think?

Reply:
Yes, I see what you mean. Wrote a small if statement.